### PR TITLE
Refactor command error messages

### PR DESCRIPTION
**Pull request by ZShunRen**
The error message for `add ` command is not specific, and does not tell the user which of their flags are missing.
In addition, the error messages for add command are too long, requiring the user to horizontally scroll to the right to read it in full.
This can reduce readability.

The proper command messages have been formatted to have more
1. line breaks
2. clear section demarkers.
The error message for `add` command is also able to inform the user which flag is missing.

Minor changes:.
1. Build.gradle file will now ignore docs, this is so that we can reduce the file size to < 100 mb.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 1]
```diff
@@ -99,34 +99,50 @@ public void parse_invalidPreamble_failure() {
...
-        // multiple invalid values, but only the first invalid value is captured
+        // multiple invalid values, all are captured
         assertParseFailure(parser, "1" + INVALID_NAME_DESC + INVALID_EMAIL_DESC
-                + VALID_ADDRESS_AMY + VALID_PHONE_AMY, Name.MESSAGE_CONSTRAINTS);
+                + VALID_ADDRESS_AMY + VALID_PHONE_AMY,
+                String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, Name.MESSAGE_CONSTRAINTS)
+                        + "\n" + Email.MESSAGE_CONSTRAINTS);
 
         // multiple invalid values but with purely optional fields, only the first in order of logical parsing is
-        // captured
```
Comment (FionaQY) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:127]: I think this line is still necessary

**Review by FionaQY — APPROVED** [review]
LGTM! Good job on the clean code and on updating all the tests.

File: src/main/java/seedu/address/logic/commands/CommandCommons.java [hunk 2]
```diff
@@ -11,4 +15,38 @@ public final class CommandCommons {
     public static final String DEFAULT_STATUS = "NONE";
     public static final Person EMPTY_PERSON = null;
 
+    /**
+     * A functional interface similar to {@link Supplier}, but allows for a checked exception, {@code ParseException}
+     * to be thrown.
+     * @param <T> The return type of the method supplied by this supplier.
+     */
+    @FunctionalInterface
+    public interface ThrowingSupplier<T> {
```
Comment (iamdiluxedbutcooler) [inline src/main/java/seedu/address/logic/commands/CommandCommons.java:24]: Nice abstraction with ThrowingSupplier!

File: build.gradle [hunk 3]
```diff
@@ -79,6 +80,17 @@ dependencies {
 
 shadowJar {
     archiveFileName = 'AgentAssist.jar'
+    zip64 = true
+    minimize()
+    exclude('**/META-INF/*.SF', '**/META-INF/*.DSA', '**/META-INF/*.RSA')
+    exclude('**/*.md') // Markdown files
+    exclude('**/*.txt') // Text files
+//    exclude('**/*.png', '**/*.jpg') // Images, if not required
```
Comment (iamdiluxedbutcooler) [inline build.gradle:88]: don't forget to clean up this commented lines later!

File: src/test/java/seedu/address/storage/JsonSerializableAgentAssistTest.java [hunk 4]
```diff
@@ -26,7 +26,6 @@ public void toModelType_typicalPersonsFile_success() throws Exception {
                 JsonSerializableAgentAssist.class).get();
         AgentAssist agentAssistFromFile = dataFromFile.toModelType();
         AgentAssist typicalPersonsAgentAssist = TypicalPersons.getTypicalAgentAssist();
-        System.out.println("Agent Assist: " + agentAssistFromFile);
```
Comment (iamdiluxedbutcooler) [inline src/test/java/seedu/address/storage/JsonSerializableAgentAssistTest.java:29]: May I ask why?

**Review by iamdiluxedbutcooler — APPROVED** [review]
LGTM! Nice work on implementing this upgrade!

File: src/test/java/seedu/address/storage/JsonSerializableAgentAssistTest.java (hunk 4 above)
Comment (ZShunRen) [reply src/test/java/seedu/address/storage/JsonSerializableAgentAssistTest.java:29]: I didn't realise you left that in as a debugging comment, let me add that back!

**Pull request merged**